### PR TITLE
Expand the block size of the root flow's margin box to the viewport size instead of expanding the block size of its border box.

### LINKED ISSUE #9308: unexpected overflow with display:inline-block + margin
rggdhwjnub

The following example makes the body element scrollable. The extra area is apparently directly related to the margin value.

Note: using `block` instead of `inline-block` reduces the scrollable area, but it's still possible to scroll.

```html
<div style="left: 0px; bottom: 0px; position: absolute;">
  <div style="margin: 200px; display: inline-block; background: red">foobar</div>
</div>
```

### PR DESCRIPTION
**Pull request by pcwalton**
Closes #9308.

r? @mbrubeck

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/9438)

### CODE REVIEW

**Comment by jdm** [comment]
@pcwalton Looks like you didn't create the new test using `./mach create-wpt`, so the manifest changes aren't correct.

**Comment by jdm** [comment]
Also the reftest file is missing the HTML element that links to the reference file.

**Comment by pcwalton** [comment]
OK, updated the test. r? @mbrubeck

File: tests/wpt/mozilla/meta/MANIFEST.json [hunk 1]
```diff
@@ -4032,6 +4032,18 @@
             "url": "/_mozilla/css/root_margin_collapse_a.html"
           }
         ],
+        "css/root_margins_a.html": [
```
Comment (jdm) [inline tests/wpt/mozilla/meta/MANIFEST.json]: This file needs a copy of this information down around line 10127.

**Comment by mbrubeck** [comment]
r=mbrubeck for the layout code.

-S-awaiting-review 

---

Reviewed 3 of 4 files at r1.
Review status: 3 of 4 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/9438#-:-K94BcsgRGLQXcoXiPls:342103440)*

**Comment by pcwalton** [comment]
OK, I updated the WPT manifest and verified that the test works.

r? @jdm

**Comment by frewsxcv** [comment]
Test expectations need to be updated

```
  ▶ PASS [expected FAIL] /css21_dev/html4/background-bg-pos-206.htm

  ▶ PASS [expected FAIL] /css21_dev/html4/bottom-applies-to-001.htm

  ▶ PASS [expected FAIL] /css21_dev/html4/bottom-applies-to-014.htm

  ▶ PASS [expected FAIL] /css21_dev/html4/bottom-applies-to-005.htm

  ▶ PASS [expected FAIL] /css21_dev/html4/bottom-applies-to-009.htm

  ▶ PASS [expected FAIL] /css21_dev/html4/bottom-applies-to-012.htm

  ▶ PASS [expected FAIL] /css21_dev/html4/bottom-applies-to-007.htm

  ▶ PASS [expected FAIL] /css21_dev/html4/bottom-applies-to-006.htm

  ▶ PASS [expected FAIL] /css21_dev/html4/bottom-applies-to-013.htm

  ▶ PASS [expected FAIL] /css21_dev/html4/bottom-applies-to-015.htm

  ▶ PASS [expected FAIL] /css21_dev/html4/containing-block-023.htm

  ▶ PASS [expected FAIL] /css21_dev/html4/position-applies-to-006.htm

  ▶ PASS [expected FAIL] /css21_dev/html4/position-applies-to-001.htm

  ▶ PASS [expected FAIL] /css21_dev/html4/position-applies-to-013.htm

  ▶ PASS [expected FAIL] /css21_dev/html4/position-applies-to-005.htm

  ▶ PASS [expected FAIL] /css21_dev/html4/position-applies-to-012.htm

  ▶ PASS [expected FAIL] /css21_dev/html4/position-applies-to-009.htm

  ▶ PASS [expected FAIL] /css21_dev/html4/position-applies-to-007.htm

  ▶ PASS [expected FAIL] /css21_dev/html4/position-applies-to-015.htm

  ▶ PASS [expected FAIL] /css21_dev/html4/position-applies-to-014.htm

```

**Comment by KiChjang** [comment]
@bors-servo retry
* #9448

**Comment by pcwalton** [comment]
Huh? About half of the Mozilla CSS tests failed. I don't think that happened locally...

**Pull request merged**